src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1206,6 +1198,16 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       cmdline.get_values("nondet-static-exclude").begin(),
       cmdline.get_values("nondet-static-exclude").end());
 
+    if(
+      cmdline.isset(FLAG_LOOP_CONTRACTS) &&
+      cmdline.isset(FLAG_LOOP_CONTRACTS_NO_UNWIND))
+    {
+      log.warning() << "**** WARNING: transformed loops will not be unwound "
+                    << "after applying loop contracts. Remember to unwind "
+                    << "them at least twice to pass unwinding-assertions."
+                    << messaget::eom;
+    }
```

Comment:
Maybe this is a discussion we've had before and I just forgot about it: can you remind me as to why we don't just unwind those loops?

Reply:
Unwinding transformed loops here will duplicate all assertions in the loops. And in some cases we don't want to duplicate assertions. For example, Kani assign every reachability check a unique ID, duplicate such assertion will result that two assertions have the same ID and fail Kani' CBMC-result rendering.
#7524

Reply:
Would you mind inserting a code comment documenting this thought process?